net-dns/dnscrypt-proxy/dnscrypt-proxy-2.0.0.ebuild

```diff
@@ -0,0 +1,89 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+EGO_PN="github.com/jedisct1/${PN}"
+
+inherit fcaps golang-build systemd user
+
+DESCRIPTION="A flexible DNS proxy, with support for encrypted DNS protocols"
+HOMEPAGE="https://github.com/jedisct1/dnscrypt-proxy"
+SRC_URI="https://${EGO_PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="ISC"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+FILECAPS=( cap_net_bind_service+ep usr/bin/dnscrypt-proxy )
+PATCHES=( "${FILESDIR}"/config-full-paths-r2.patch )
+
+pkg_setup() {
+	enewgroup dnscrypt
+	enewuser dnscrypt -1 -1 /var/empty dnscrypt
+}
+
+src_compile() {
+	# Create directory structure suitable for building
+	mkdir -p "src/${EGO_PN}" || die
+	rm -r "src/${EGO_PN}" || die
+	mv "${PN}" "src/${EGO_PN}" || die
+	mv "vendor" "src/" || die
+	golang-build_src_compile
+}
+
+src_install() {
+	default
+
+	dobin dnscrypt-proxy
+
+	insinto /etc/${PN}
+	newins "src/${EGO_PN}"/example-dnscrypt-proxy.toml dnscrypt-proxy.toml
+	doins "src/${EGO_PN}"/example-{blacklist.txt,cloaking-rules.txt,forwarding-rules.txt}
+
+	insinto "/usr/share/${PN}"
+	doins -r "utils/generate-domains-blacklists/"
+
+	newinitd "${FILESDIR}"/${PN}.initd-r2 ${PN}
+	newconfd "${FILESDIR}"/${PN}.confd-r2 ${PN}
+	systemd_newunit "${FILESDIR}"/${PN}.service-r2 ${PN}.service
+	systemd_dounit systemd/dnscrypt-proxy.socket
+
+	keepdir /var/log/${PN}
+}
+
+pkg_postinst() {
+	fcaps_pkg_postinst
+
+	if ! use filecaps; then
+		ewarn
```

Comment:
Please remove the preceding empty lines, so you don't end up having two empty lines between each section.